lib/tlscontext.c

```diff
@@ -697,6 +675,50 @@ tls_context_load_key_and_cert(TLSContext *self)
   return TLS_CONTEXT_OK;
 }
 
+static void _write_line_to_keylog_file_mutex(const char *file_path, const char *line)
+{
+  FILE *keylog_file;
+  gint ret_val;
+  static GStaticMutex mutex;
```

Comment:
The lock is global this way, which will block all TLS key writes, even if they are in separate keylog files.


I think this should go to `TLSContext` as a field, for example: `keylog_file_lock`.
This would be safe, because every source/destination driver has a separate keylog file, and in case of sources, all TLS connections from the same driver shares a single `TLSContext` instance.